pandas/core/base.py

```diff
@@ -1,10 +1,13 @@
 """
 Base and utility classes for pandas objects.
 """
+from typing import Any, Callable, Optional, Tuple, Union
```

Comment:
instead of doing this directly, move this to ``pandas.types.hinting.py`` and import from there. IOW that will be the 1 place that we inherit typing things from.

we can then handle the compat issues when ``typing`` is not installed (on 2.7).